common/src/main/java/whocraft/tardis_refined/common/tardis/TardisDesktops.java

```diff
@@ -1,38 +1,75 @@
+    public static final DesktopTheme FACTORY_THEME = new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "factory"), new ResourceLocation(TardisRefined.MODID, "desktop/factory"), true);
 
-    public static final DesktopTheme FACTORY_THEME = registerDesktop(new DesktopTheme("factory", new ResourceLocation(TardisRefined.MODID, "desktop/factory"), true));
-    public static final DesktopTheme CORAL_THEME = registerDesktop(new DesktopTheme("coral", new ResourceLocation(TardisRefined.MODID, "desktop/coral"), true));
-    public static final DesktopTheme TOYOTA_THEME = registerDesktop(new DesktopTheme("toyota", new ResourceLocation(TardisRefined.MODID, "desktop/toyota"), true));
-    public static final DesktopTheme COPPER = registerDesktop(new DesktopTheme("copper", new ResourceLocation(TardisRefined.MODID, "desktop/copper"), true));
-    public static final DesktopTheme CRYSTAL = registerDesktop(new DesktopTheme("crystal", new ResourceLocation(TardisRefined.MODID, "desktop/crystal"), true));
-    public static final DesktopTheme VICTORIAN = registerDesktop(new DesktopTheme("victorian", new ResourceLocation(TardisRefined.MODID, "desktop/victorian"), true));
-    public static final DesktopTheme NUKA = registerDesktop(new DesktopTheme("nuka", new ResourceLocation(TardisRefined.MODID, "desktop/nuka"), true));
-    public static final DesktopTheme FUTURE_NOSTALGIA = registerDesktop(new DesktopTheme("future_nostalgia", new ResourceLocation(TardisRefined.MODID, "desktop/future_nostalgia"), true));
+    public TardisDesktops() {
+        super(TardisRefined.GSON, "desktop");
+    }
 
 
-    public static DesktopTheme registerDesktop(DesktopTheme theme) {
-        DESKTOPS.add(theme);
-        return theme;
+    public static DesktopTheme getDesktopById(ResourceLocation location) {
+        return DESKTOPS.stream().filter(theme -> Objects.equals(theme.identifier, location)).findAny().or(() -> Optional.of(FACTORY_THEME)).get();
     }
 
-    public static DesktopTheme getDesktopThemeById(String id) {
-        return DESKTOPS.stream().filter(theme -> Objects.equals(theme.id, id)).findAny().or(() -> Optional.of(FACTORY_THEME)).get();
+    public static void clear() {
+        DESKTOPS.clear();
     }
 
+    @Override
+    protected void apply(Map<ResourceLocation, JsonElement> object, ResourceManager resourceManager, ProfilerFiller profilerFiller) {
+        DESKTOPS.clear();
+        object.forEach((resourceLocation, jsonElement) -> {
+            try {
+
+                JsonObject desktopObject = jsonElement.getAsJsonObject();
+
+                // Build the DesktopTheme
+                String id = desktopObject.get("id").getAsString();
+                String structure = desktopObject.get("structure").getAsString();
+                String name = desktopObject.get("name_component").getAsString();
+                ResourceLocation resourceLocationId = new ResourceLocation(id);
```

Comment:
The "available by default" field doesn't seem to be part of the JSON format. Is that intended? 

If it's not meant to be available at this time, please add a comment just so we don't forget later on.

Reply:
Looking at this more, AvaliableByDefault doesn't need to exist anymore. I've removed it from the system entirely.